openmetadata-service/src/main/java/org/openmetadata/service/events/ChangeEventHandler.java

```diff
@@ -242,6 +242,17 @@ private List<Thread> getThreads(ContainerResponseContext responseContext, String
       return List.of(thread);
     }
 
+    int responseCode = responseContext.getStatus();
```

Comment:
1. there are too many if conditions which can easily turned into switch or if else conditions
2. Wouldn't this add too many entity created , especially incase of ingestion, we are filtering on ingestion-bot so hopefully this shouldn't be an issue cc @mohityadav766 
3.  can't we keep the if condition simple like this RestUtil.ENTITY_SOFT_DELETED.equals(changeType)? instead of adding responseCode and not sure why we are checking  !responseContext.getEntity().getClass().equals(Thread.class)) . This should be filtered above itself no?